stable/rabbitmq-ha/templates/secret.yaml

```diff
@@ -19,12 +19,12 @@ data:
   {{ else }}
   rabbitmq-erlang-cookie: {{ randAlphaNum 32 | b64enc | quote }}
   {{ end }}
-{{- if .Values.rabbitmqCert.enabled }}
+{{- if and .Values.rabbitmqCert.enabled (or .Values.rabbitmqCert.cacertfile .Values.rabbitmqCert.certfile .Values.rabbitmqCert.keyfile) }}
```

Comment:
I don't like this logic. This should still only depend on `rabbitmqCert.enabled`. Otherwise this flag wouln't make sense.

Reply:
I don't agree. `rabbitmqCert.enabled` triggers whether we need to include the certificate in the `volumeMounts` section of the `StatefulSet`. If we're using a certificate _and_ passing in the values of the certificates, we need this `Secret`. However, if we are using a certificate that already exists as a secret, we won't be passing in `cacertfile` or `certfile` or `keyfile` and we don't need this `Secret`.

Reply:
Actually after I make the change you suggested below to change `secretName` to `existingSecret` I can change this trigger to be based on that and it will be cleaner since having `existingSecret` set means we don't need this secret.

Reply:
I think it's better to still include `.Values.rabbitmqCert.enabled` in the condition.

Reply:
👍 Good call. Included `.Values.rabbitmqCert.enabled` in the check as well.